### PR TITLE
[Ehn] plot fov

### PR DESCRIPTION
**Pull request by mts299**
# Scope

This code encapsulates just the field of view (FOV) plotting code to be further used for `plot_grid` and `plot_map`

Does not change the functionality of `plot_fan`

## Test

```python
import matplotlib.pyplot as plt 
import datetime as dt
import pydarn

#Read in fitACF file
file = "20131031.2201.00.sas.fitacf"
SDarn_read = pydarn.SuperDARNRead(file)
fitacf_data = SDarn_read.read_fitacf()

fanplot = pydarn.Fan.plot_fov(66, dt.datetime(2013,10,31,22,1), fov_color='b')
fanplot = pydarn.Fan.plot_fov(fitacf_data[0]['stid'], dt.datetime(2013,10,31,22,1), fov_color='r')
#Make the fan plot and show
plt.show()
```

plotfov

Please note I am aware of the lines gonna try to fix that but also give people a chance to generally test the code

### CODE REVIEW

**Review by Devin-Huyghebaert — APPROVED** [review]
Tested: MCM, SAS, INV, FHW, HOK

The colors follow the outline of the FOV with the recent updates.

Warning output when running:
fan.py: 294: UserWarning: FixedFormatter should only be used together with FixedLocator

Should default lower latitude be consistent with changes made for pyDARN 2.0 PR?  It was changed to +/- 30 degrees I believe.

The code was not checked in this review, though the output looks good.

File: pydarn/plotting/fan.py [hunk 1]
```diff
@@ -238,9 +214,115 @@ def plot_fan(cls, dmap_data: List[dict], ax=None, scan_index: int = 1,
...
 
             if colorbar_label != '':
                 cb.set_label(colorbar_label)
 
-        return beam_corners_aacgm_lats, beam_corners_aacgm_lons, scan, grndsct, dtime
+        return beam_corners_aacgm_lats, beam_corners_aacgm_lons, scan, grndsct
+
+    @classmethod
+    def plot_fov(cls, stid: str, dtime: dt.datetime, ax=None,
+                 lowlat: int = 50, ranges: List = [0, 75],
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:227]:
```suggestion
                 lowlat: int = 30, ranges: List = [0, 75],
```
following Devins suggestion

File: pydarn/plotting/fan.py [hunk 2]
```diff
@@ -238,9 +214,115 @@ def plot_fan(cls, dmap_data: List[dict], ax=None, scan_index: int = 1,
...
+        # Setup plot
+        # This may screw up references
+        if ax is None:
+            ax = plt.axes(polar=True)
+            if beam_corners_aacgm_lats[0, 0] > 0:
+                ax.set_ylim(90, lowlat)
+                ax.set_yticks(np.arange(lowlat, 90, 10))
+            else:
+                ax.set_ylim(-90, -lowlat)
+                ax.set_yticks(np.arange(-lowlat, -90, -10))
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:293]:
```suggestion
                ax.set_ylim(-90, -abs(lowlat))
                ax.set_yticks(np.arange(-abs(lowlat), -90, -10))
```
Same changes in fan plotting review

**Review by carleyjmartin — APPROVED** [review]
I just made suggestions to keep the code the same as the fan plot defaults, and the changes to make it able to plot a given lowlat in the southern hemisphere. Other than those it looks great. 

Tested with following code for SAS, INV, RKN, PGR and CLY:
```python
fanplot = pydarn.Fan.plot_fov(66, dt.datetime(2016,11,23,16,1), fov_color='k')
fanplot = pydarn.Fan.plot_fov(64, dt.datetime(2016,11,23,16,1), fov_color='r')
fanplot = pydarn.Fan.plot_fov(65, dt.datetime(2016,11,23,16,1), fov_color='b')
fanplot = pydarn.Fan.plot_fov(5, dt.datetime(2016,11,23,16,1), fov_color='g')
fanplot = pydarn.Fan.plot_fov(6, dt.datetime(2016,11,23,16,1), fov_color='y')
plt.show()
```
Screen Shot 2021-02-23 at 9 19 55 AM

and southern hemisphere with MCM:
```python
file = "/Users/carley/Documents/data/20161123.1601.00.mcm.a.fitacf"
SDarn_read = pydarn.SuperDARNRead(file)
fitacf_data = SDarn_read.read_fitacf()

fanplot = pydarn.Fan.plot_fov(fitacf_data[0]['stid'], dt.datetime(2016,11,23,10,1), fov_color='k')
fanplot = pydarn.Fan.plot_fov(fitacf_data[0]['stid'], dt.datetime(2016,11,23,12,1), fov_color='r')
fanplot = pydarn.Fan.plot_fov(fitacf_data[0]['stid'], dt.datetime(2016,11,23,14,1), fov_color='b')
fanplot = pydarn.Fan.plot_fov(fitacf_data[0]['stid'], dt.datetime(2016,11,23,16,1), fov_color='g')
fanplot = pydarn.Fan.plot_fov(fitacf_data[0]['stid'], dt.datetime(2016,11,23,18,1), fov_color='y')
plt.show()
```
Screen Shot 2021-02-23 at 9 26 10 AM

I don't know if I missed it on the fan plots, but the axes lines in grey are over the top of the fill but underneath the black edges. With the fill being transparent it might look better if it's underneath that too. 

I get warnings:
```
294: UserWarning: FixedFormatter should only be used together with FixedLocator
287: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
```

Is that second warning going to be an issue if we want multiple fans?

**Comment by mts299** [comment]
So I am going to take issue with the "userWarning" from matplotlib. It's not the same warning as before but it is tricky to handle since we are using polar plots. 

Gonna merge the code since we got 2 approvals now and addressed all other issues.

**Pull request merged**